articles/rosbags.md

```diff
@@ -0,0 +1,185 @@
+It should further be possible to configure the condition on when to write such a chunk permanently to disk (e.g. in a given time interval or when chunk size is reached).
+
+#### Backwards compatible with ROS1
+A general requirement is to be backwards compatible with existing ROS1 bags.
+This compatibility can either be via a conversion script, which permanently converts ROS1 bags into ROS2 bags or a bridge API which allows to manually open existing ROS1 bags and publish them into the ROS2 system.
+
+
+### Dismissed data storage formats
+
+In the following, we are iterating over a couple of data formats, which may be suitable for the underlying ROSbag implementation.
+We hereby iterate over existing third party software as well as examining of maintaining a self-made format.
+
+
+#### HDF5
+
+One very popular framework for storing scientific data is [HDF5](https://support.hdfgroup.org/HDF5/).
+It basically has all the necessary requirements listed above such as random access, parallelism and compression.
+It is further designed for highly complex data with an extensive amount of data.
+HDF5 is open source and its source can be freely obtained from [bitbucket](https://bitbucket.hdfgroup.org/projects/HDFFV/repos/hdf5/browse) and is under a [permissive license](https://bitbucket.hdfgroup.org/projects/HDFFV/repos/hdf5/browse/COPYING) of the HDF group.
+Multiple language wrapper or bindings are available, namely C/C++, python, fortran or java.
+
+##### Pros
+- Open Source and standard specification for file format
+- Fulfills the requirements given
+- Multi language support
+- Large community of users
+
+##### Cons
+- Depending on a slowly developing standard
+- The table dimensions of each chunk have to be of fixed size, known at startup.
+
+There is a popular [blog post](http://cyrille.rossant.net/moving-away-hdf5/) by Cyrille Rossant, which gives a short introduction, but also discusses some controversy with HDF5.
+
+
+#### Existing ROS1 format
+
+Alternatively, the existing ROS1 format can be continued to be used. The format description can be found [here](http://wiki.ros.org/Bags/Format/2.0).
+
+##### Pros
+- Already ROS specific and evaluated for ROS messages, existing code could be reused
+
+##### Cons
+- No random access
+
+
+#### SQLite
+
+A third alternative which provides capabilities for data logging, and is used as such, is [SQLite](https://www.sqlite.org/about.html)
+Despite other relational database systems, it doesn't require any external SQL Server, but is self-contained.
+It is also open source, [extensively tested](https://www.sqlite.org/testing.html) and known to have a large [community](https://www.sqlite.org/support.html).
+The Gazebo team created a nicely composed [comparison](https://osrfoundation.atlassian.net/wiki/spaces/GAZ/pages/99844178/SQLite3+Proposal) between SQLite and the existing rosbag format.
```

Comment:
RTI's [recording tools](https://community.rti.com/static/documentation/connext-dds/5.3.0/doc/manuals/recording_service/RTI_Recording_Service_UsersManual.pdf) are also based on SQLite, and seem to work pretty well.  You might want to experiment with them as part of your research.